From: Alexander Block <ablock84@googlemail.com>
Date: Sat, 28 Jul 2012 16:32:45 +0200
Patch-mainline: 3.7
Git-commit: 52f9e53ede8e1b261e68216c6c2f32bb3f26c795
References: FATE#312888
Subject: [PATCH] Btrfs: use normal return path for root == send_root
 case

Don't have a seperate return path for the mentioned case. Now
we do the same "take lowest inode/offset" logic for all found clones.

Commit is a result of Arne's review.

Signed-off-by: Alexander Block <ablock84@googlemail.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/send.c |    6 ------
 1 file changed, 6 deletions(-)

--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -1109,12 +1109,6 @@ static int __iterate_backrefs(u64 ino, u
 			return 0;
 		if (offset + ctx->extent_len > ctx->cur_offset)
 			return 0;*/
-
-		bctx->found++;
-		found->found_refs++;
-		found->ino = ino;
-		found->offset = offset;
-		return 0;
 	}
 
 	bctx->found++;
